### PR TITLE
Repository infrastructure setup

### PR DESCRIPTION
**Pull request by scalvert**
This PR adds initial repository setup, including

- A workspaces architecture for multi-package support
- A workspaces typescript configuration
- Basic top-level setup
	- npm scripts
	- repo-wide devDependencies
	- root-level eslint/prettier setup to keep packages consistent
	- [`volta`](https://volta.sh/) to manage tool dependencies
- A skeleton of the `sarif-builder` package

All infrastructure choices and naming conventions are open to feedback and changes.

### CODE REVIEW

**Comment by scalvert** [comment]
cc/ @jeffersonking - you don't seem to be added as a contributor to the repo yet.

**Comment by jeffersonking** [comment]
@scalvert Re yarn vs npm, it sounds like `release-it` will be the deciding factor. Will try to test it out sometime.

**Comment by scalvert** [comment]
@jeffersonking correct. The [current plugin](https://github.com/rwjblue/release-it-yarn-workspaces) operates on yarn workspaces, and hasn't been tested with npm. That said, we own that plugin and can update/adjust it to work with both.

**Comment by jeffersonking** [comment]
@scalvert If all else is equal, the path with one less installation wins? So `npm`?

**Comment by scalvert** [comment]
Yep, we can certainly explore either making that plugin work for npm, or make a similar plugin for npm.

**Comment by scalvert** [comment]
Looking at our plugin, I think it will work out-of-the-box, as it's not doing anything yarn specific. It likely could use a rename though...

**Comment by scalvert** [comment]
OK I've updated to use `npm` vs. `yarn`. I also pinned tool dependencies with [`volta`](https://volta.sh/). If you aren't familiar with it, it's a tool that manages your project's tool dependencies.

**Comment by jeffersonking** [comment]
@scalvert Thanks! Not familiar with `volta`, interested in trying it out.

I think the there's still the big question of if the `sarif-builder` goes in it's own repo (and if anything ends up in this repo). But, signing off on this PR nevertheless.

**Comment by scalvert** [comment]
Ya I think the question of 'what else goes here' is for sure up for debate, but I can imagine the following:

- Moving the types from `@types/sarif` to this repository, effectively making them first-party types vs. third-party
- A SARIF log validator utility
- A partial fingerprint utility to enable SARIF log authors to generate consistent fingerprints
- Any additional documentation or reference material related specifically to using SARIF in JavaScript.

Happy to defer these discussions until they present themselves later.

**Review by jeffersonking — APPROVED** [review]

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      ]
+    }
+  },
+  "env": {
+    "browser": false,
+    "node": true,
+    "es6": true
+  },
+  "rules": {
+    "prettier/prettier": "error",
```
Comment (rwjblue) [inline .eslintrc:35]: Can you swap this to use `plugin:prettier/recommended` to the end of `extends`?

Comment (scalvert) [reply .eslintrc]: Yep, done.

**Comment by scalvert** [comment]
@jeffersonking `volta` is pretty great. It was developed at LinkedIn by folks from my team. It is gaining a lot of traction as a replacement to nvm, mainly due to speed since it's written in Rust.

**Review by rwjblue — APPROVED** [review]

**Comment by michaelcfanning** [comment]
Wahoo! Every GitHub open source journey starts with a single merge!

**Pull request merged**